src/test/java/com/cegeka/academy/repository/GroupUserRoleRepositoryTest.java

```diff
@@ -0,0 +1,102 @@
+package com.cegeka.academy.repository;
+
+import com.cegeka.academy.AcademyProjectApp;
+import com.cegeka.academy.domain.Group;
+import com.cegeka.academy.domain.GroupUserRole;
+import com.cegeka.academy.domain.Role;
+import com.cegeka.academy.domain.User;
+import com.cegeka.academy.repository.util.TestsRepositoryUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@SpringBootTest(classes = AcademyProjectApp.class)
+@Transactional
+public class GroupUserRoleRepositoryTest {
+
+    @Autowired
+    private GroupUserRoleRepository groupUserRoleRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    private User user1, user2;
+    private Group group;
+    private GroupUserRole groupUserRole1, groupUserRole2;
+    private Role role;
+
+    @BeforeEach
+    public void init() {
+
+        user1 = TestsRepositoryUtil.createUser("login1", "1aaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaana");
+        userRepository.saveAndFlush(user1);
+        user2 = TestsRepositoryUtil.createUser("login2", "2ananaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaana");
+        userRepository.saveAndFlush(user2);
+        group = TestsRepositoryUtil.createGroup("gr1", "ana are mere");
+        groupRepository.save(group);
+        role = TestsRepositoryUtil.createRole("admin");
+        roleRepository.save(role);
+        groupUserRole1 = TestsRepositoryUtil.createGroupUserRole(user1, groupRepository.findAll().get(0), roleRepository.findAll().get(0));
+        groupUserRoleRepository.save(groupUserRole1);
+        groupUserRole2 = TestsRepositoryUtil.createGroupUserRole(user2, groupRepository.findAll().get(0), roleRepository.findAll().get(0));
+        groupUserRoleRepository.save(groupUserRole2);
+    }
+
+    @Test
+    public void testSaveGroupUserRole() {
+
+        List<GroupUserRole> list = groupUserRoleRepository.findAll();
+        assertThat(list.size()).isEqualTo(2);
+        assertThat(list.get(0).getGroup()).isEqualTo(groupRepository.findAll().get(0));
```

Comment:
Pozitia unui `User` sau `Group` in listele returnate de `findAll` este relativa deci nu ar trebui sa te bazezi pe indecsi ci ar trebui sa cauti entitatea in lista.